src/main/java/net/minecraftforge/client/animation/EntityAnimation.java

```diff
@@ -0,0 +1,173 @@
+     * <li>PASSIVE - DEFAULT</li>
+     * <li>PASSIVE - LAST</li>
+     * <li>ACTIVE - FIRST</li>
+     * <li>ACTIVE - DEFAULT (Cancelled if an ACTIVE - FIRST is executed)</li>
+     * <li>ACTIVE - LAST (Cancelled if an ACTIVE - FIRST/DEFAULT is executed)</li>
+     * </ul>
+     * @param o an entity animation
+     * @return See {@link Comparable} documentation
+     */
+    @Override
+    public final int compareTo(@NotNull EntityAnimation<E> o)
+    {
+        if (this.getMode() == o.getMode())
+        {
+            return this.getPriority().ordinal() - o.getPriority().ordinal();
+        }
+        else if (o.getMode() == Mode.ACTIVE)
+        {
+            return -1;
+        }
+        return 1;
+    }
+
+    /**
+     * The animation mode. Read individual enums for an explanation on which one to use.
+     */
+    public enum Mode
+    {
+        /**
+         * A passive animation is considered a background animation like walking or sitting. It is
+         * executed first before active animations. It can override or be overridden by other
+         * passive animations.
+         */
+        PASSIVE,
+
+        /**
+         * An active animation is considered an important animation. It will override passive
+         * animations. The big difference is that once an active animation is executed, all remaining
+         * active animations will not be executed.
+         */
+        ACTIVE
+    }
+
+    /**
+     * The priority of the animation. Read individual enums for an explanation on which one to use.
+     */
+    public enum Priority
+    {
+        /**
+         * Marks the animation to execute first. If the animation mode is {@link Mode#PASSIVE}, it
+         * may be overridden by animations with the same or later priority. Otherwise if the mode is
+         * {@link Mode#ACTIVE} and the animation is executed, active animations with the same or
+         * later priority will not be executed.
+         */
+        FIRST,
+
+        /**
+         * The default priority of animations. A passive animation executed with this priority can
+         * override any passive animations executed in an earlier priority. It is not guaranteed that
+         * an active animation will be executed at this priority if one was executed in an early
+         * priority.
+         */
+        DEFAULT,
+
+        /**
+         * Marks the animation to run last. A passive animation executed with this priority can
+         * override any passive animations executed in an earlier priority. It is not guaranteed that
+         * an active animation will be executed at this priority if one was executed in an early
+         * priority.
+         */
+        LAST
+    }
+
+    /**
+     * A record containing data used when calculating animations with the addition of partial ticks
+     */
+    protected record Context(float animateTicks, float animateSpeed, float bobAnimateTicks, float headYaw, float headPitch, float partialTicks) {}
```

Comment:
Please put the brace on a newline. Also, I believe the convention in Mojang code is that this is called the 'partial tick' (since it's a fraction of a single tick).

Reply:
Fixed and changed